pyccl/halos/profiles_2pt.py

```diff
@@ -25,32 +28,31 @@ class Profile2pt(CCLAutoreprObject):
     """
     __repr_attrs__ = ("r_corr",)
 
-    def __init__(self, r_corr=0.):
+    @warn_api
+    def __init__(self, *, r_corr=0.):
         self.r_corr = r_corr
 
     __eq__ = object.__eq__
 
     __hash__ = object.__hash__  # TODO: remove once __eq__ is replaced.
 
-    def update_parameters(self, r_corr=None):
+    @warn_api
+    def update_parameters(self, *, r_corr=None):
         """ Update any of the parameters associated with this 1-halo
         2-point correlator. Any parameter set to `None` won't be updated.
         """
         if r_corr is not None:
             self.r_corr = r_corr
 
-    def fourier_2pt(self, prof, cosmo, k, M, a,
-                    prof2=None, mass_def=None):
+    @warn_api
+    def fourier_2pt(self, cosmo, k, M, a, prof, *, prof2=None, mass_def):
```

Comment:
The arguments have been reordered, I believe. Will this preserve API?

Reply:
yes

Reply:
Yes, because the posiitional arguments have been reordered (not the keyword ones) `warn_api` specifically checks for the type of the first argument to see if the old API or the new API has been used, and reorders accordingly. That's the only place we reorder posiitonal arguments, and now that I'm looking at it again, I don't even know if this is a good idea. See my proposal about absorbing that into the `HaloProfile` base class.